src/dapp/components/ui/HalveningBanner.tsx

```diff
@@ -1,24 +1,56 @@
+import "./Banner.css";
+
+import { useService } from "@xstate/react";
 import React, { useState } from "react";
 
-import alert from "../../images/ui/expression_alerted.png";
 import closeIcon from "../../images/ui/close.png";
-
-import "./Banner.css";
+import alert from "../../images/ui/expression_alerted.png";
+import { service } from "../../machine";
+import { numberWithBreaks } from "../../utils/number";
+import { getNextHalvingThreshold } from "../../utils/supply";
 
 export const Banner: React.FC = () => {
   const [show, setShow] = useState(true);
+  const [totalSupply, setTotalSupply] = React.useState<number>(0);
+
+  const [machineState, send] = useService<
+    Context,
+    BlockchainEvent,
+    BlockchainState
+  >(service);
+
+  React.useEffect(() => {
+    const load = async () => {
+      const notReady =
+        machineState.matches("loading") ||
+        machineState.matches("initial") ||
+        machineState.matches("registering") ||
+        machineState.matches("creating") ||
+        machineState.matches("onboarding");
+
+      if (notReady && totalSupply < 1) setTimeout(() => load(), 100);
+
+      const supply =
+        await service.machine.context.blockChain.totalSupply();
+
+      setTotalSupply(supply);
+    };
+
+    load();
+  }, [machineState]);
+
+  if (!show || !totalSupply) return null;
 
-  if (!show) {
-    return null;
-  }
+  const nextHalvingThreshold = getNextHalvingThreshold(totalSupply);
 
   return (
     <div id="halvening-banner">
       <img src={alert} />
       <div>
         <span>
-          When total supply reaches 1 Million, crop and upgrade prices will be
-          divided by 1/5. Be prepared!
+          When total supply reaches{" "}
+          {numberWithBreaks(nextHalvingThreshold)}, crop and upgrade prices
+          will be divided by 1/5. Be prepared!
```

Comment:
Each halvening period alternates between 1/2 and 1/5.

For instance the next halvening will be 1/2. Consider adding some logic to display the correct division as well :)